core/src/main/java/org/elasticsearch/index/mapper/internal/ParentFieldMapper.java

```diff
@@ -223,20 +236,36 @@ public Query termsQuery(List values, @Nullable QueryParseContext context) {
     }
 
     private final String type;
+    private final MappedFieldTypeReference joinFieldTypeRef;
+    private final MappedFieldType defaultJoinFieldType;
 
-    protected ParentFieldMapper(MappedFieldType fieldType, String type, Settings indexSettings) {
-        super(NAME, setupDocValues(indexSettings, fieldType), setupDocValues(indexSettings, Defaults.FIELD_TYPE), indexSettings);
+    protected ParentFieldMapper(MappedFieldType fieldType, MappedFieldType joinFieldType, String type, Settings indexSettings) {
+        super(NAME, fieldType, Defaults.FIELD_TYPE, indexSettings);
         this.type = type;
+        this.joinFieldTypeRef = new MappedFieldTypeReference(joinFieldType);
+        this.defaultJoinFieldType = Defaults.JOIN_FIELD_TYPE;
+    }
+
+    public ParentFieldMapper(Settings indexSettings, MappedFieldType existing, String parentType) {
+        this(existing == null ? Defaults.FIELD_TYPE.clone() : existing.clone(), joinFieldTypeForParentType(parentType, indexSettings), null, indexSettings);
     }
 
-    public ParentFieldMapper(Settings indexSettings, MappedFieldType existing) {
-        this(existing == null ? Defaults.FIELD_TYPE.clone() : existing.clone(), null, indexSettings);
+    private static MappedFieldType joinFieldTypeForParentType(String parentType, Settings indexSettings) {
```

Comment:
We should look at requiring this to be explicit in the future. I think it would be much cleaner to have to say "this is a parent" and "this is a child", instead of just the latter. But ok for now.